src/Components/ImagesTable/Instance.tsx

```diff
@@ -0,0 +1,209 @@
+            aria-label="Open launch wizard"
+          >
+            <ProvisioningWizard
+              onClose={() => openWizard(false)}
+              image={{
+                name: compose.image_name || compose.id,
+                id: compose.id,
+                architecture:
+                  compose.request.image_requests[0].upload_request.options,
+                provider: provider,
+                sourceIDs: sourceIds,
+                accountIDs: accountIds,
+                uploadOptions:
+                  compose.request.image_requests[0].upload_request.options,
+                uploadStatus: composeStatus.image_status.upload_status,
+              }}
+            />
+          </Modal>
+        )}
+      </>
+    );
+  }
+};
+
+const getImageProvider = (compose: ComposesResponseItem) => {
+  const imageType = compose.request.image_requests[0].image_type;
+  switch (imageType) {
+    case 'aws':
+      return 'aws';
+    case 'ami':
+      return 'aws';
+    case 'azure':
+      return 'azure';
+    case 'gcp':
+      return 'gcp';
+    default:
+      //TODO check with Provisioning: what if imageType is not 'aws', 'ami', or 'azure'?
+      return 'aws';
+  }
+};
+
+type AwsS3InstancePropTypes = {
+  compose: ComposesResponseItem;
+  composeStatus: ComposeStatus;
+  isExpired: boolean;
+};
+
+export const AwsS3Instance = ({
+  compose,
+  composeStatus,
+  isExpired,
+}: AwsS3InstancePropTypes) => {
+  const navigate = useNavigate();
+
+  const fileExtensions: { [key in ImageTypes]: string } = {
+    aws: '',
+    azure: '',
+    'edge-commit': '',
+    'edge-installer': '',
+    gcp: '',
+    'guest-image': '.qcow2',
+    'image-installer': '.iso',
+    vsphere: '.vmdk',
+    'vsphere-ova': '.ova',
+    wsl: '.tar.gz',
+    ami: '',
+    'rhel-edge-commit': '',
+    'rhel-edge-installer': '',
+    vhd: '',
+  };
+
+  const status = composeStatus.image_status.status;
+  const options = composeStatus.image_status.upload_status?.options;
+
+  if (options && !isAwss3UploadStatus(options)) {
+    throw TypeError(
+      `Error: options must be of type Awss3UploadStatus, not ${typeof options}.`
+    );
+  }
+
+  if (status === 'success' && !isExpired) {
+    return (
+      <Button
+        component="a"
+        target="_blank"
+        variant="link"
+        isInline
+        href={options?.url}
+      >
+        Download ({fileExtensions[compose.request.image_requests[0].image_type]}
+        )
+      </Button>
+    );
+  } else if (status === 'success' && isExpired) {
+    return (
+      <Button
+        component="a"
+        target="_blank"
+        variant="link"
+        onClick={() => navigate(resolveRelPath(`imagewizard/${compose.id}`))}
+        isInline
+      >
+        Recreate image
+      </Button>
+    );
+  } else {
+    return (
+      <Button isDisabled variant="link" isInline>
+        Download ({fileExtensions[compose.request.image_requests[0].image_type]}
+        )
+      </Button>
+    );
+  }
+};
```

Comment:
Nitpick:

I would prefer

```
if(status === 'success'){
  if(isExpired){
    return ...
  }
  return ...
}
return ...
```

Reply:
Good point, I fixed it but in a slightly different way to avoid nesting:

```
  if (status !== 'success') {
    ...
  } else if (!isExpired) {
    ...
  } else if (isExpired) {
    ...
  }

```